src/System.Net.Http/tests/FunctionalTests/SocketsHttpHandlerTest.cs

```diff
@@ -1141,6 +1141,51 @@ public async Task SmallConnectionTimeout_SubsequentRequestUsesDifferentConnectio
             }
         }
 
+        [Theory]
+        [InlineData("PooledConnectionLifetime")]
+        [InlineData("PooledConnectionIdleTimeout")]
+        public async Task Http2_SmallConnectionTimeout_SubsequentRequestUsesDifferentConnection(string timeoutPropertyName)
```

Comment:
I was looking into that but it was tricky. Some of the low-level function don't work quite the same.
Also to get it working we would need to overload that class with UseHttp2 and there are tests which do not need to run again (like range checks) 

Also I realized overnight that existing code is wrong as the client should send GOAWAY and we should verify that. 

cc: @davidsh

Reply:
I check with libhttp2 and it does not send GOAWAY either. Since we are expiring connection only when all streams are gone it may be ok to keep it as is. This can possibly be revisited as part of #34638